core/eolearn/core/utils/parsing.py

```diff
@@ -139,11 +171,14 @@ def _parse_dict(self, features: dict) -> List[Union[Tuple[FeatureType, str, str]
         return feature_specs
 
     def _parse_sequence(
-        self, features: Sequence
-    ) -> List[Union[Tuple[FeatureType, str, str], Tuple[FeatureType, None, None]]]:
-        """Implements parsing and validation in case the input is a sequence."""
+        self,
+        features: Union[
+            Tuple[FeatureType, str], Tuple[FeatureType, str, str], Tuple[FeatureType, EllipsisType], SequenceFeatureSpec
```

Comment:
Can't this now be `Union[SingleFeatureSpec, SequenceFeatureSpec]`? There are some differences, but I though that bbox/timestamp cases in a tuple with `None` could in principle also be expected, or not?

Reply:
Good catch, there is a potential issue here indeed. Mypy for some reason refines the type to just `SequenceFeatureSpec` and ignores the possibility of Tuples. Will need to figure out a way to clarify that :thinking:

Reply:
I switched to your suggestion. Mypy continues to wrongly refine the type, but that is a bug which i reported to the mypy repo. No easy way to circumvent things, `Iterable` and `Sequence` are apparently giving them headaches with type refinement.